controllers/machinedeployment_sync.go

```diff
@@ -349,6 +350,14 @@ func (r *MachineDeploymentReconciler) scale(ctx context.Context, deployment *clu
 // syncDeploymentStatus checks if the status is up-to-date and sync it if necessary.
 func (r *MachineDeploymentReconciler) syncDeploymentStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet, d *clusterv1.MachineDeployment) error {
 	d.Status = calculateStatus(allMSs, newMS, d)
+
+	minReplicasNeeded := *(d.Spec.Replicas) - mdutil.MaxUnavailable(*d)
+	if d.Status.AvailableReplicas >= minReplicasNeeded {
+		// NOTE: The structure of calculateStatus() does not allow us to update the machinedeployment directly, we can only update the status obj it returns. Ideally, we should change calculateStatus() --> updateStatus() to be consistent with the rest of the code base, until then, we update conditions here.
+		conditions.MarkTrue(d, clusterv1.AvailableCondition)
+	} else {
+		conditions.MarkFalse(d, clusterv1.AvailableCondition, clusterv1.MinimumMachinesAvailableReason, clusterv1.ConditionSeverityWarning, "MachineDeployment does not have minimum availability of %d replicas, current availability is %d", minReplicasNeeded, d.Status.AvailableReplicas)
```

Comment:
```suggestion
		conditions.MarkFalse(d, clusterv1.AvailableCondition, clusterv1.MinimumMachinesAvailableReason, clusterv1.ConditionSeverityWarning, "Minimum availability requires %d replicas, current %d available", minReplicasNeeded, d.Status.AvailableReplicas)
```
Let's avoid MachineDeployment given that those conditions are already inside the MD  object.